### PR TITLE
Bot logic complete

### PR DESCRIPTION
**Pull request by SaurusXI**
Now we need to create a GUI that reacts to the events fired by the bot.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  "description": "<p align=\"center\"> <a href=\"https://dscvit.com\"> \t<img src=\"https://user-images.githubusercontent.com/30529572/72455010-fb38d400-37e7-11ea-9c1e-8cdeb5f5906e.png\" alt=\"DSC VIT\"/> </a> \t<h2 align=\"center\"> Airbender </h2> \t<h4 align=\"center\"> Draw and highlight user avatars in Discord Voice Channel <h4> </p>",
+  "main": "index.js",
+  "dependencies": {
+    "discord.js": "^12.3.1",
+    "opusscript": "^0.0.7",
+    "typescript": "^4.0.2"
+  },
+  "devDependencies": {},
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (L04DB4L4NC3R) [inline package.json:13]: Run script should be here as `start`.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/SaurusXI/airbender.git"
+  },
+  "author": "",
+  "license": "MIT",
```
Comment (L04DB4L4NC3R) [inline package.json]: GPL LICENCE

File: bot/speech_event.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const { EventEmitter } = require('events');
```
Comment (L04DB4L4NC3R) [inline bot/speech_event.js:1]: Redundant import

File: bot/message_handler.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+function handleMessage(msg) {
```
Comment (L04DB4L4NC3R) [inline bot/message_handler.js]: ES5 function used here. Use something like this:

`const handleMessage = (msg) => {}`

File: bot/listener.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+function listen(conn, sc) {
```
Comment (L04DB4L4NC3R) [inline bot/listener.js]: ES5 function used

File: bot/index.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+client.on('ready', async () => {
+  console.log(`Logged in as ${client.user.tag}!`);
+});
+
+client.on('message', async message => {
+  let msg = handleMessage(message);
+  if (!msg.valid) return;
+
+  if (msg.instruction === 'join') {
```
Comment (L04DB4L4NC3R) [inline bot/index.js]: Pass instruction to an external function which has switch case

File: bot/index.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
+const Discord = require('discord.js');
+
+const handleMessage = require('./message_handler').handleMessage;
```
Comment (L04DB4L4NC3R) [inline bot/index.js]: Use destructuring

File: bot/index.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+const listenerFunc = require('./listener').listenerFunc;
```
Comment (L04DB4L4NC3R) [inline bot/index.js]: Use destructuring

File: bot/index.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+var speechConnection = require('./speech_event').speechConnection;
```
Comment (L04DB4L4NC3R) [inline bot/index.js]: Use destructuring and const

File: bot/index.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+      if (message.member.voice.channel) {
+          message.member.voice.channel.leave();
+    }
+  }
+});
+
+client
+    .login(process.env.DISCORD_TOKEN)
+    .catch(error => {
+        console.log(error);
```
Comment (L04DB4L4NC3R) [inline bot/index.js]: Use `console.error`

File: bot/listener.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    console.log("Joined voice channel successfully");
+    console.log("Play join.mp3");
+
+    // For the speaking event to fire more than once, 
+    // an audio file must be played in the channel
+    // see - https://github.com/discordjs/discord.js/issues/3524
+    dispatcher = conn.play("audio/sound.mp3", { passes: 5 });
```
Comment (L04DB4L4NC3R) [inline bot/listener.js]: Initialize local dispatcher with `let`

File: bot/message_handler.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+    };
+
+    if (msg.author.bot) return result;
+
+    if (!msg.content.startsWith(process.env.PREFIX)) return result;
+
+    console.log("Message is valid");
+    result.valid = true;
+
+    const args = msg.content.slice(process.env.PREFIX.length).trim().split(/ +/g);
```
Comment (L04DB4L4NC3R) [inline bot/message_handler.js]: This much chaining should have a nextline

**Review by L04DB4L4NC3R — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (SaurusXI) [reply package.json:13]: The run script will spawn the GUI - since that isn't made yet, I haven't mentioned any scripts in this section

**Comment by SaurusXI** [comment]
@L04DB4L4NC3R please check now

**Review by L04DB4L4NC3R — APPROVED** [review]

**Pull request merged**